src/js/server/kettleValidation.js

```diff
@@ -0,0 +1,107 @@
+    var toValidate = fluid.model.transformWithRules(requestHandler.req, transformationRules);
+    var validationResults = globalValidator.validate(gssSchema, toValidate, schemaHash);
+
+    if (validationResults.isValid) {
+        requestHandler.handleValidRequest(requestHandler);
+    }
+    else {
+        var localisedErrors = gpii.schema.validator.localiseErrors(validationResults.errors, toValidate, kettleValidator.model.messages, kettleValidator.options.localisationTransform);
+        var localisedPayload = fluid.copy(validationResults);
+        localisedPayload.errors = localisedErrors;
+
+        var failurePayload = fluid.extend({}, kettleValidator.options.errorTemplate, localisedPayload);
+        requestHandler.events.onError.fire(failurePayload);
+    }
+};
+
+gpii.schema.kettle.validator.cacheSchemas = function (kettleValidator, globalValidator) {
+    fluid.each(kettleValidator.options.requestSchemas, function (requestSchema, schemaKey) {
+        var schemaHash = gpii.schema.hashSchema(requestSchema);
+        kettleValidator.schemaHashes[schemaKey] = schemaHash;
+        globalValidator.cacheSchema(requestSchema, schemaHash);
+    });
+};
+
+fluid.defaults("gpii.schema.kettle.validator", {
+    gradeNames: ["fluid.modelComponent"],
+    members: {
+        schemaHashes: {}
+    },
+    model: {
+        messages: gpii.schema.messages.validationErrors
+    },
+    localisationTransform: {
+        "": ""
+    },
+    errorTemplate: {
+        // "Bad Request": https://developer.mozilla.org/nl/docs/Web/HTTP/Status/400
+        statusCode: 400,
+        message: "Your request was invalid.  See the errors for details."
+    },
+    requestSchemas: {
+        default: {
+            "$schema": "gss-v7-full#"
+        }
+    },
+    requestContentToValidate: {
+        default: "{that}.options.requestContentToValidate.body",
+        body: {
+            "": "body"
+        },
+        params: {
+            "": "params"
+        },
+        query: {
+            "": "query"
+        }
+    },
+    invokers: {
+        validateRequest: {
+            funcName: "gpii.schema.kettle.validator.validateRequest",
+            args: ["{that}", "{gpii.schema.validator}", "{arguments}.0"] // kettleValidator, globalValidator, request
+        }
+    },
+    listeners: {
+        "onCreate.cacheSchemas": {
+            funcName: "gpii.schema.kettle.validator.cacheSchemas",
+            args:     ["{that}", "{gpii.schema.validator}"] // globalValidator
+        }
+    }
+});
+
+fluid.defaults("gpii.schema.kettle.app", {
+    gradeNames: ["kettle.app"],
+    components: {
+        validator: {
+            type: "gpii.schema.kettle.validator"
+        }
+    }
+});
+
+fluid.defaults("gpii.schema.kettle.request.http", {
+    gradeNames: ["kettle.request.http"],
+    rulesKey: "default",
+    schemaKey: "default",
+    invokers: {
+        handleRequest: {
```

Comment:
I don't think we want to intrude on the native request handling workflow of the target component in this way. The original concept was for the validator to be packaged as a piece of statically available middleware in the "eternal".